### ISSUE TITLE
Chapter 9 Notes, Errata, and Discussion: Self-Contained Scala Scripts

### ISSUE THREAD

**Issue by lihaoyi** [handsonscala/handsonscala#9]
Thread to discuss anything and everything about Chapter 9. Important topics will be folded into the issue description.

**Comment by implisci** [comment]
// The timestamp code won't compile on Java 8
// value ofInstant is not a member of object java.time.LocalDate
```scala
val publishDate = java.time.LocalDate.ofInstant(
        java.time.Instant.ofEpochMilli(os.mtime(p)),
        java.time.ZoneOffset.UTC
      )
```

// This works with Java 8
```scala
val publishDate = java.time.Instant.ofEpochMilli(os.mtime(p)).atZone(java.time.ZoneOffset.UTC).toLocalDate()
```

**Comment by tim8west** [comment]
Possible typo page 164 in bash terminal output. Should _posts_ read simply _post_?

**Comment by dlqs** [comment]
typo nit: https://github.com/handsonscala/handsonscala/blob/v1/snippets/9.3.bash#L7
```
$ ls
Blog.sc	posts
```
`posts` should be `post`

**Comment by dlqs** [comment]
https://github.com/handsonscala/handsonscala/blob/v1/snippets/9.24.scala#L12
`os.proc("git", "push", targetGitRepo, "head", "-f").call(cwd = os.pwd / "out")`
Should this be `master`/`HEAD` instead of `head`? There's no ref called head by default.

**Comment by ijk0** [comment]
https://github.com/handsonscala/handsonscala/blob/v1/examples/9.1%20-%20Printing/Blog.sc
```
method s is not a case class, nor does it have an unapply/unapplySeq member
    val s"$prefix - $suffix.md" = p.last
        ^
Compilation Failed
```
hi, when I run this snippt,  there came out the comlile error.

**Comment by lihaoyi** [comment]
@ijk0 you are probably on an old version of Scala or old version of Ammonite. You need to be on Scala 2.13.x for pattern matching on strings

**Comment by lihaoyi** [comment]
@dlqs yeah it should be `HEAD`. Lowercase `head` works on Mac OSX since the filesystem isn't case sensitive, but on Linux you need to use uppercase `HEAD`
